src/pages/invites/index.tsx

```diff
@@ -0,0 +1,236 @@
+// implement show all invites from local as list ordered from new to old
+import { FC, useState } from 'react';
+import {
+  deleteInviteLocally,
+  Invite,
+  makeInviteUrl,
+  updateInviteLocally,
+} from '@utils/invite';
+import Blockies from 'react-blockies';
+import Key from '@media/UI/invite/key.png';
+import NextImage from '@media/UI/invite/next.png';
+import BackImage from '@media/UI/invite/back.png';
+import EditImage from '@media/UI/invite/pencil.png';
+import SaveImage from '@media/UI/invite/check-mark.png';
+import DeleteImage from '@media/UI/invite/delete.png';
+import DollarImage from '@media/UI/invite/dollar-sign.png';
+import EmptyImage from '@media/UI/invite/empty.png';
+import copy from 'copy-to-clipboard';
+
+interface AllInvitesProps {
+  invites: Invite[];
+  updateInvites: () => void;
+  onTopUpInvite: (invite: Invite) => void;
+}
+
+export const ITEMS_PER_PAGE = 6;
+
+const Invites: FC<AllInvitesProps> = ({
+  invites,
+  updateInvites,
+  onTopUpInvite,
+}) => {
+  const [currentPage, setCurrentPage] = useState(0);
+  // keep track of the invite being edited
+  const [editingInvite, setEditingInvite] = useState<Invite>(null);
+  const [editingInviteName, setEditingInviteName] = useState('');
+  const [hoverInviteId, setHoverInviteId] = useState<string | null>(null);
+
+  const handleEditClick = (invite: Invite) => {
+    setEditingInvite(invite);
+    setEditingInviteName(invite.name || invite.id);
+  };
+
+  const handleDeleteClick = (invite: Invite) => {
+    if (!window.confirm('Are you sure you want to delete this invite?')) {
```

Comment:
Using the native dialogs is a bad UI practice. It's better to use some of the existing modals like `ConfirmDeleteModal`.